File: server/game/cards/01-Core/ANobleCause.js
import PlotCard from '../../plotcard.js';

class ANobleCause extends PlotCard {
    setupCardAbilities(ability) {
        this.persistentEffect({
            targetController: 'current',
            effect: ability.effects.reduceFirstMarshalledCardCostEachRound(
                2,
                (card) => card.hasTrait('Lord') || card.hasTrait('Lady')
            )
        });
    }
}

ANobleCause.code = '01004';

export default ANobleCause;

File: throneteki-v2/src/Throneteki.Cards/Implementations/Packs/CoreSet/ANobleCause.cs
using Throneteki.Cards.Abilities;

namespace Throneteki.Cards.Implementations.Packs.CoreSet;

/// <summary>
/// A Noble Cause (01004) — Income 3, Initiative 6, Claim 1, Reserve 6.
/// When Revealed: Reduce the cost of the next Lord or Lady you marshal this
/// phase by 2 (simplified: no abilities, cost reduction handled by effect engine).
/// </summary>
[CardDefinition("01004")]
public sealed class ANobleCause : CardScript
{
    // Cost reduction is a persistent effect registered at reveal time
    // by the effect engine. No card script ability needed.
}
